code/services/QueuedJobService.php

```diff
@@ -615,14 +615,20 @@ public function runJob($jobId) {
 
 		Config::unnest();
 
-		// okay lets reset our user if we've got an original
+		// okay let's reset our user if we've got an original
 		if ($runAsUser && $originalUser) {
 			Session::clear("loggedInAs");
 			if ($originalUser) {
 				Session::set("loggedInAs", $originalUser->ID);
 			}
 		}
 
+		// let's make sure there is a cleanupJob in the queue
+		if (Config::inst()->get('CleanupJob', 'is_enabled')) {
```

Comment:
If your deployment process is something that takes 10s of minutes across multiple distributed backend nodes, that requires a convoluted approval process to perform, I'd much prefer being able to simply remove it from the UI and not have it restart rather than go through that if I need to stop it from running.